### DISCUSSION TITLE
Mouse trajectory/position

### DISCUSSION THREAD

**Discussion by GaiaCo** [jspsych/jsPsych#1289] [General]
Hello, 
I am looking for a way to record mouse trajectory during a html-button-resonse. I found ways to do so in JavaScript but have no idea how to integrate it into the plug in. 
Has anyone ever done this/knows how to? 

every bit of help is highly appreciated!
Thank you!

**Comment by becky-gilbert** [comment]
Hi @GaiaCo! This seems like something that someone else might've put together already, but I can't find any posts about it in this Discussion board or in the old Google Group. Hopefully if someone has done this already, they'll chime in and share their code!

We're [planning to add mouse tracking to the 7.1 release](https://github.com/jspsych/jsPsych/discussions/908), but I'm not sure if that works for your timeline. Also note that we're behind schedule right now 😬 

If you've already found a way of doing this in JavaScript, then it might not be too difficult to turn that into a jsPsych plugin. There's a [page on the jsPsych website about making plugins](https://www.jspsych.org/plugins/creating-a-plugin/), and Josh made a [video tutorial about making plugins](https://youtu.be/XQcsFwAmbiw) too. 

Also, Alex Irvine has shared a ['mouse view' jsPsych plugin](https://github.com/u01ai11/MouseView.js#jspsych) that (1) hides the page content, (2) tracks/records mouse movements, and (3) creates a viewable area around the current mouse position. This is overkill for what you want to do, but I thought I'd let you know in case you want to modify it so that it just does the mouse tracking.

If you want to share your code here then maybe we can offer some more specific feedback.

**Reply by GaiaCo** [reply]
Thank you for getting back to me! I'll try to create a new plug in and if I make it I'll be sure to post it here.  

Thanks!

**Reply by revatishivnekar** [reply]
Hey GaiaCo!
Did you ever get around to making that plugin? :)

**Comment by GEJ1** [comment]
Hi @GaiaCo , I made a plugin for draw and mouse tracking that can be modified for your purposes: https://github.com/jspsych/jsPsych/discussions/1370 

I think that the long term solution it would be to implement an [extension](https://www.jspsych.org/extensions/extensions/) for mouse tracking.
